syntax = "proto3";

enum Genre {
    GENRE_UNDEFINED = 0;
    GENRE_FANTASY = 1;
    GENRE_SCIENCE_FRICTION = 2;
    GENRE_MYSTERY = 3;
    GENRE_HORROR = 4;
}

message Book {
  string ISBN = 1;
  string title = 2;
  string author = 3;
  Genre genre = 4;
  int32 publishingYear = 5;
}

enum Status {
    AVAILABLE = 0;
    TAKEN = 1;
}

message InventoryItem {
    int32 inventoryNumber = 1;
    oneof items {
        Book book = 2;
    }
    Status status = 3;
}

message ISBN {
    string ISBN = 1;
}

enum StatusCode {
    UNDEFINED = 0;
    SUCCESS = 1;
    FAILURE = 2;
}

message CreateBookResponseStatus {
    StatusCode StatusCode = 1;
    string message = 2;
}

message GetBookResponseStatus {
    StatusCode statusCode = 1;
    string message = 2;
    Book book = 3;
}

